std/math.d

```diff
@@ -7468,14 +7471,77 @@ if (isFloatingPoint!(F) && isIntegral!(G))
             return 1.0;
         case 1:
             return x;
-        case 2:
-            return x * x;
         default:
         }
 
         v = x;
     }
 
+    // Bail out early, if we can estimate that the result is infinity or 0.0:
+    //
+    // We use the following two conclusions:
+    //
+    //    m * floor(log2(abs(v))) >= F.max_exp
+    // =>             abs(v) ^^ m >  F.max == nextDown(F.infinity)
+    //
+    //    m * (bias - ex - 1) >= bias + F.mant_dig - 1
+    // =>         abs(v) ^^ m <  2 ^^ (-bias - F.mant_dig + 2) == nextUp(0.0)
+    //
+    // floor(log2(abs(v))) == ex - bias can be directly taken from the
+    // exponent of the floating point represantation, to avoid long
+    // calculations here.
+
+    enum uint bias = F.max_exp - 1;
+
+    static if (is(F == float))
+    {
+        float f = cast(float) v;
+        uint ival = () @trusted { return *cast(uint*) &f; }();
+        ulong ex = (ival >> 23) & 255;
+    }
+    else static if (is(F == double) || (is(T == real) && T.mant_dig == double.mant_dig))
```

Comment:
There's no `T` here. Even when changing that to `F`, I'm seeing a unittest regression with LDC on Windows, with 64-bit `real`.

Reply:
IMHO it wouldn't have done any harm. These `static ifs` only provide shortcut for cases, where you do know ahead the result. If they aren't compiled in, the slower calculation at the bottom of the function is used. With the typo this essentially is `else static if (is(F == double) || false)`. And that's exactly the fix I suggest: Just use `else static if (is(F == double)`, which just means: Don't use the speed up for those 64bit-reals.

Due to personal reasons I'll not going to write this PR. If anyone feels inclined, please go on.